assets/js/components/adminbar/common-GA4.stories.js

```diff
@@ -0,0 +1,232 @@
+	{
+		startDate: '2020-12-31',
+		endDate: '2021-01-27',
+		compareStartDate: '2020-12-03',
+		compareEndDate: '2020-12-30',
+		metrics: [
+			{
+				name: 'totalUsers',
+			},
+		],
+		url: 'https://www.sitekitbygoogle.com/blog/',
+	},
+
+	// Mock options for mocking Sessions report's response.
+	{
+		startDate: '2020-12-31',
+		endDate: '2021-01-27',
+		compareStartDate: '2020-12-03',
+		compareEndDate: '2020-12-30',
+		dimensions: [
+			{
+				name: 'date',
+			},
+		],
+		limit: 10,
+		metrics: [
+			{
+				name: 'sessions',
+			},
+		],
+		url: 'https://www.sitekitbygoogle.com/blog/',
+	},
+];
+
+export const setupBaseRegistry = ( registry, args ) => {
+	// Set some site information.
+	provideSiteInfo( registry, {
+		currentEntityURL: 'https://www.sitekitbygoogle.com/blog/',
+		currentEntityTitle: 'Blog test post for Google Site Kit',
+	} );
+
+	// Set up analytics and analytics-4 modules stores but provide no data.
+	provideModules( registry, [
+		{
+			slug: 'analytics',
+			active: true,
+			connected: true,
+		},
+		{
+			slug: 'analytics-4',
+			active: true,
+			connected: true,
+		},
+	] );
+
+	// Call story-specific setup.
+	if ( typeof args?.setupRegistry === 'function' ) {
+		args.setupRegistry( registry );
+	}
+};
+
+export const widgetDecorators = [
+	( Story ) => (
+		<div className="googlesitekit-widget">
+			<div className="googlesitekit-widget__body">
+				<Story />
+			</div>
+		</div>
+	),
+	( Story, { args } ) => {
+		const setupRegistry = ( registry ) => {
+			setupBaseRegistry( registry, args );
+		};
+
+		return (
+			<WithRegistrySetup func={ setupRegistry }>
+				<Story />
+			</WithRegistrySetup>
+		);
+	},
+];
+
+export const setupAnalytics4MockReports = (
+	registry,
+	mockOptions = adminbarAnalytics4OptionSets
+) => {
+	registry.dispatch( CORE_USER ).setReferenceDate( '2021-01-28' );
+	registry.dispatch( MODULES_ANALYTICS_4 ).setPropertyID( '1000' );
+	mockOptions.forEach( ( options ) => {
+		registry
+			.dispatch( MODULES_ANALYTICS_4 )
+			.receiveGetReport( getAnalytics4MockResponse( options ), {
+				options,
+			} );
+	} );
+};
+
+export const setupAnalytics4GatheringData = (
+	registry,
+	mockOptionSets = adminbarAnalytics4OptionSets
+) => {
+	registry.dispatch( CORE_USER ).setReferenceDate( '2021-01-28' );
+
+	const propertyID = properties[ 0 ]._id;
+	// Set the property creation timestamp to one and a half days ago, so that
+	// the property is considered to be in the gathering data state.
+	const createTime = new Date(
+		Date.now() - DAY_IN_SECONDS * 1.5 * 1000
+	).toISOString();
+
+	const property = {
+		...properties[ 0 ],
+		createTime,
+	};
+	registry
+		.dispatch( MODULES_ANALYTICS_4 )
+		.receiveGetProperty( property, { propertyID } );
+	registry.dispatch( MODULES_ANALYTICS_4 ).setPropertyID( propertyID );
+	mockOptionSets.forEach( ( options ) => {
+		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetReport(
+			{
+				rows: [],
+				totals: [ {}, {} ],
+			},
+			{
+				options,
+			}
+		);
```

Comment:
The GA4 API returns an empty report in the same format as the zero data variant when in the gathering data state - it doesn't have a `rows` property at all.

This could be rewritten to use `replaceValuesInAnalytics4ReportWithZeroData()` similar to the zero data story below. We may need to revisit this in future depending on how #6623 pans out, but this would be accurate as things stand.

```suggestion
		const report = getAnalyticsMockResponse( options );
		const zeroReport =
			replaceValuesInAnalytics4ReportWithZeroData( report );
		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetReport( zeroReport, {
			options,
		} );
```

Reply:
Updated.